## `nixpkgs-review` result

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 472525`
Commit: [`c47efdc28a55b6a7ef3a2a962d8806e5429f48fd`](https://github.com/NixOS/nixpkgs/commit/c47efdc28a55b6a7ef3a2a962d8806e5429f48fd) ([subsequent changes](https://github.com/NixOS/nixpkgs/compare/c47efdc28a55b6a7ef3a2a962d8806e5429f48fd..pull/472525/head))
Merge: [`9c64a3470b56b653ff5084553520fe048203bef6`](https://github.com/NixOS/nixpkgs/commit/9c64a3470b56b653ff5084553520fe048203bef6)

Logs: https://github.com/cybardev/nixpkgs-review-action/actions/runs/20383987360


---
### `x86_64-linux`
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>docker_25</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 29 packages built:</summary>
  <ul>
    <li>airlift</li>
    <li>airlift.dist</li>
    <li>arion</li>
    <li>arion.data</li>
    <li>charliecloud</li>
    <li>copacetic</li>
    <li>devcontainer</li>
    <li>docker (docker_29)</li>
    <li>docker-client</li>
    <li>docker-compose</li>
    <li>docker-gc</li>
    <li>docker-language-server</li>
    <li>docker-sbom</li>
    <li>docker-vackup</li>
    <li>docker_28</li>
    <li>fn-cli</li>
    <li>gefyra</li>
    <li>langgraph-cli (python313Packages.langgraph-cli)</li>
    <li>langgraph-cli.dist (python313Packages.langgraph-cli.dist)</li>
    <li>pipework</li>
    <li>python312Packages.jupyter-repo2docker</li>
    <li>python312Packages.jupyter-repo2docker.dist</li>
    <li>python312Packages.langgraph-cli</li>
    <li>python312Packages.langgraph-cli.dist</li>
    <li>python313Packages.jupyter-repo2docker</li>
    <li>python313Packages.jupyter-repo2docker.dist</li>
    <li>tests.devShellTools.nixos</li>
    <li>winboat</li>
    <li>wrkflw</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `x86_64-linux`</summary>
<details>
<summary>docker_25</summary>
<pre>Using versionCheckHook
Running phase: unpackPhase
@nix { &quot;action&quot;: &quot;setPhase&quot;, &quot;phase&quot;: &quot;unpackPhase&quot; }
unpacking source archive /nix/store/lay8p8rcab41ibxj3ng5gnrf4y1cj2qf-source
source root is source
Running phase: patchPhase
@nix { &quot;action&quot;: &quot;setPhase&quot;, &quot;phase&quot;: &quot;patchPhase&quot; }
applying patch /nix/store/dcxskwwpi4v3mzhkfinfv5aiy5rx0w62-cli-system-plugin-dir-from-env.patch
patching file cli-plugins/manager/manager_unix.go
Hunk #1 FAILED at 2.
Hunk #2 FAILED at 12.
2 out of 2 hunks FAILED -- saving rejects to file cli-plugins/manager/manager_unix.go.rej</pre>
</details>
</details>

---
### `aarch64-linux`
<details>
  <summary>:x: 2 packages failed to build:</summary>
  <ul>
    <li>docker_25</li>
    <li>tests.devShellTools.nixos</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 27 packages built:</summary>
  <ul>
    <li>airlift</li>
    <li>airlift.dist</li>
    <li>arion</li>
    <li>arion.data</li>
    <li>charliecloud</li>
    <li>copacetic</li>
    <li>devcontainer</li>
    <li>docker (docker_29)</li>
    <li>docker-client</li>
    <li>docker-compose</li>
    <li>docker-gc</li>
    <li>docker-language-server</li>
    <li>docker-sbom</li>
    <li>docker-vackup</li>
    <li>docker_28</li>
    <li>fn-cli</li>
    <li>gefyra</li>
    <li>langgraph-cli (python313Packages.langgraph-cli)</li>
    <li>langgraph-cli.dist (python313Packages.langgraph-cli.dist)</li>
    <li>pipework</li>
    <li>python312Packages.jupyter-repo2docker</li>
    <li>python312Packages.jupyter-repo2docker.dist</li>
    <li>python312Packages.langgraph-cli</li>
    <li>python312Packages.langgraph-cli.dist</li>
    <li>python313Packages.jupyter-repo2docker</li>
    <li>python313Packages.jupyter-repo2docker.dist</li>
    <li>wrkflw</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `aarch64-linux`</summary>
<details>
<summary>docker_25</summary>
<pre>Using versionCheckHook
Running phase: unpackPhase
@nix { &quot;action&quot;: &quot;setPhase&quot;, &quot;phase&quot;: &quot;unpackPhase&quot; }
unpacking source archive /nix/store/lay8p8rcab41ibxj3ng5gnrf4y1cj2qf-source
source root is source
Running phase: patchPhase
@nix { &quot;action&quot;: &quot;setPhase&quot;, &quot;phase&quot;: &quot;patchPhase&quot; }
applying patch /nix/store/dcxskwwpi4v3mzhkfinfv5aiy5rx0w62-cli-system-plugin-dir-from-env.patch
patching file cli-plugins/manager/manager_unix.go
Hunk #1 FAILED at 2.
Hunk #2 FAILED at 12.
2 out of 2 hunks FAILED -- saving rejects to file cli-plugins/manager/manager_unix.go.rej</pre>
</details>
</details>

---
### `x86_64-darwin` (sandbox = relaxed)
<details>
  <summary>:x: 2 packages failed to build:</summary>
  <ul>
    <li>copacetic</li>
    <li>docker_25</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 22 packages built:</summary>
  <ul>
    <li>airlift</li>
    <li>airlift.dist</li>
    <li>arion</li>
    <li>arion.data</li>
    <li>devcontainer</li>
    <li>docker (docker-client, docker_29)</li>
    <li>docker-compose</li>
    <li>docker-gc</li>
    <li>docker-language-server</li>
    <li>docker-sbom</li>
    <li>docker-vackup</li>
    <li>docker_28</li>
    <li>fn-cli</li>
    <li>langgraph-cli (python313Packages.langgraph-cli)</li>
    <li>langgraph-cli.dist (python313Packages.langgraph-cli.dist)</li>
    <li>python312Packages.jupyter-repo2docker</li>
    <li>python312Packages.jupyter-repo2docker.dist</li>
    <li>python312Packages.langgraph-cli</li>
    <li>python312Packages.langgraph-cli.dist</li>
    <li>python313Packages.jupyter-repo2docker</li>
    <li>python313Packages.jupyter-repo2docker.dist</li>
    <li>wrkflw</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `x86_64-darwin`</summary>
<details>
<summary>copacetic</summary>
<pre>Building subPackage ./test/e2e/multiplatform-plugin
github.com/project-copacetic/copacetic/test/e2e/multiplatform-plugin: no non-test Go files in /nix/build/nix-8636-1724484074/source/test/e2e/multiplatform-plugin
Building subPackage ./test/e2e/plugin
github.com/project-copacetic/copacetic/test/e2e/plugin: no non-test Go files in /nix/build/nix-8636-1724484074/source/test/e2e/plugin
Building subPackage ./test/e2e/push
github.com/project-copacetic/copacetic/test/e2e/push: no non-test Go files in /nix/build/nix-8636-1724484074/source/test/e2e/push
buildPhase completed in 6 minutes 52 seconds
Running phase: checkPhase
@nix { &quot;action&quot;: &quot;setPhase&quot;, &quot;phase&quot;: &quot;checkPhase&quot; }
ok  	github.com/project-copacetic/copacetic/integration/multiarch	1.326s [no tests to run]
ok  	github.com/project-copacetic/copacetic/integration/singlearch	1.811s [no tests to run]
time=&quot;2025-12-19T22:49:50Z&quot; level=warning msg=&quot;Running on macOS, assuming Docker Desktop handles emulation.&quot;
time=&quot;2025-12-19T22:49:50Z&quot; level=warning msg=&quot;Running on macOS, assuming Docker Desktop handles emulation.&quot;
time=&quot;2025-12-19T22:49:50Z&quot; level=warning msg=&quot;Running on macOS, assuming Docker Desktop handles emulation.&quot;
--- FAIL: TestQemuAvailable_Mocked (0.04s)
    --- FAIL: TestQemuAvailable_Mocked/no_match_at_all (0.01s)
        buildkit_test.go:512: QemuAvailable() = true, want false
FAIL
FAIL	github.com/project-copacetic/copacetic/pkg/buildkit	1.666s
FAIL</pre>
</details>
<details>
<summary>docker_25</summary>
<pre>Using versionCheckHook
Running phase: unpackPhase
@nix { &quot;action&quot;: &quot;setPhase&quot;, &quot;phase&quot;: &quot;unpackPhase&quot; }
unpacking source archive /nix/store/lay8p8rcab41ibxj3ng5gnrf4y1cj2qf-source
source root is source
Running phase: patchPhase
@nix { &quot;action&quot;: &quot;setPhase&quot;, &quot;phase&quot;: &quot;patchPhase&quot; }
applying patch /nix/store/dcxskwwpi4v3mzhkfinfv5aiy5rx0w62-cli-system-plugin-dir-from-env.patch
patching file cli-plugins/manager/manager_unix.go
Hunk #1 FAILED at 2.
Hunk #2 FAILED at 12.
2 out of 2 hunks FAILED -- saving rejects to file cli-plugins/manager/manager_unix.go.rej</pre>
</details>
</details>

---
### `aarch64-darwin` (sandbox = relaxed)
<details>
  <summary>:x: 2 packages failed to build:</summary>
  <ul>
    <li>copacetic</li>
    <li>docker_25</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 22 packages built:</summary>
  <ul>
    <li>airlift</li>
    <li>airlift.dist</li>
    <li>arion</li>
    <li>arion.data</li>
    <li>devcontainer</li>
    <li>docker (docker-client, docker_29)</li>
    <li>docker-compose</li>
    <li>docker-gc</li>
    <li>docker-language-server</li>
    <li>docker-sbom</li>
    <li>docker-vackup</li>
    <li>docker_28</li>
    <li>fn-cli</li>
    <li>langgraph-cli (python313Packages.langgraph-cli)</li>
    <li>langgraph-cli.dist (python313Packages.langgraph-cli.dist)</li>
    <li>python312Packages.jupyter-repo2docker</li>
    <li>python312Packages.jupyter-repo2docker.dist</li>
    <li>python312Packages.langgraph-cli</li>
    <li>python312Packages.langgraph-cli.dist</li>
    <li>python313Packages.jupyter-repo2docker</li>
    <li>python313Packages.jupyter-repo2docker.dist</li>
    <li>wrkflw</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `aarch64-darwin`</summary>
<details>
<summary>copacetic</summary>
<pre>Building subPackage ./test/e2e/multiplatform-plugin
github.com/project-copacetic/copacetic/test/e2e/multiplatform-plugin: no non-test Go files in /nix/build/nix-8239-808362120/source/test/e2e/multiplatform-plugin
Building subPackage ./test/e2e/plugin
github.com/project-copacetic/copacetic/test/e2e/plugin: no non-test Go files in /nix/build/nix-8239-808362120/source/test/e2e/plugin
Building subPackage ./test/e2e/push
github.com/project-copacetic/copacetic/test/e2e/push: no non-test Go files in /nix/build/nix-8239-808362120/source/test/e2e/push
buildPhase completed in 2 minutes 19 seconds
Running phase: checkPhase
@nix { &quot;action&quot;: &quot;setPhase&quot;, &quot;phase&quot;: &quot;checkPhase&quot; }
ok  	github.com/project-copacetic/copacetic/integration/multiarch	0.017s [no tests to run]
ok  	github.com/project-copacetic/copacetic/integration/singlearch	0.020s [no tests to run]
time=&quot;2025-12-19T22:29:11Z&quot; level=warning msg=&quot;Running on macOS, assuming Docker Desktop handles emulation.&quot;
time=&quot;2025-12-19T22:29:11Z&quot; level=warning msg=&quot;Running on macOS, assuming Docker Desktop handles emulation.&quot;
time=&quot;2025-12-19T22:29:11Z&quot; level=warning msg=&quot;Running on macOS, assuming Docker Desktop handles emulation.&quot;
--- FAIL: TestQemuAvailable_Mocked (0.00s)
    --- FAIL: TestQemuAvailable_Mocked/no_match_at_all (0.00s)
        buildkit_test.go:512: QemuAvailable() = true, want false
FAIL
FAIL	github.com/project-copacetic/copacetic/pkg/buildkit	0.022s
FAIL</pre>
</details>
<details>
<summary>docker_25</summary>
<pre>Using versionCheckHook
Running phase: unpackPhase
@nix { &quot;action&quot;: &quot;setPhase&quot;, &quot;phase&quot;: &quot;unpackPhase&quot; }
unpacking source archive /nix/store/lay8p8rcab41ibxj3ng5gnrf4y1cj2qf-source
source root is source
Running phase: patchPhase
@nix { &quot;action&quot;: &quot;setPhase&quot;, &quot;phase&quot;: &quot;patchPhase&quot; }
applying patch /nix/store/dcxskwwpi4v3mzhkfinfv5aiy5rx0w62-cli-system-plugin-dir-from-env.patch
patching file cli-plugins/manager/manager_unix.go
Hunk #1 FAILED at 2.
Hunk #2 FAILED at 12.
2 out of 2 hunks FAILED -- saving rejects to file cli-plugins/manager/manager_unix.go.rej</pre>
</details>
</details>

--- a/pkgs/applications/virtualization/docker/compose.nix
+++ b/pkgs/applications/virtualization/docker/compose.nix
@@ -2,6 +2,7 @@
   lib,
   buildGoModule,
   fetchFromGitHub,
+  versionCheckHook,
 }:
 
 buildGoModule rec {
@@ -17,17 +18,20 @@ buildGoModule rec {
 
   vendorHash = "sha256-EFbEd1UwrBnH6pSh+MvupYdie8SnKr8y6K9lQflBSlk=";
 
+  nativeInstallCheckInputs = [ versionCheckHook ];
+
   modPostBuild = ''
     patch -d vendor/github.com/docker/cli/ -p1 < ${./cli-system-plugin-dir-from-env.patch}
   '';
 
   ldflags = [
-    "-X github.com/docker/compose/v2/internal.Version=${version}"
+    "-X github.com/docker/compose/v5/internal.Version=${version}"
     "-s"
     "-w"
   ];
 
   doCheck = false;
+  doInstallCheck = true;
   installPhase = ''
     runHook preInstall
     install -D $GOPATH/bin/cmd $out/libexec/docker/cli-plugins/docker-compose

